jib-core/src/main/java/com/google/cloud/tools/jib/image/json/JsonToImageTranslator.java

```diff
@@ -115,5 +134,33 @@
     return imageBuilder.build();
   }
 
+  /**
+   * Converts a map of exposed ports as strings to a list of {@link Port}s (e.g. {@code
+   * {"1000/tcp":{}}} -> {@code Port(1000, Protocol.TCP)}).
+   *
+   * @param portMap the map to convert
+   * @return a list of {@link Port}s
+   */
+  @VisibleForTesting
+  static ImmutableList<Port> portMapToList(@Nullable Map<String, Map<?, ?>> portMap)
+      throws BadConfigurationFormatException {
+    if (portMap == null) {
+      return ImmutableList.of();
+    }
+    ImmutableList.Builder<Port> ports = new ImmutableList.Builder<>();
+    for (Map.Entry<String, Map<?, ?>> entry : portMap.entrySet()) {
+      String port = entry.getKey();
+      Matcher matcher = portPattern.matcher(port);
```

Comment:
Would this be better on `Port` (though I suppose this could be considered a file-specific format)? Or breaking out to a separate method?

Reply:
I think we wanted to avoid translation to/from file-specific formats in `Port` since it's going to be a user-facing class in the jib library, and we don't want to break users if the format needed to change.

Maybe we could break this out to a separate method, but this is the only place it's used, so I think it's something we could do later if necessary (similar code is in `ExposedPortParser`, but again I think we wanted to differentiate between parsing plugin configuraiton and parsing container configuration, even if the formats are similar right now).

Reply:
That makes sense, I suppose.  By plugin-configuration, you mean for Maven/Gradle plugin configuration, right?

Reply:
Yeah, Maven/Gradle in our case, and however users choose to parse their input when jib becomes a library.